index.html

```diff
@@ -204,55 +205,2367 @@ <h3>
+        [=uses report IDs=], the [=byte sequence=] MUST NOT include the report
+        ID byte.
+        </li>
+        <li>[=Fire an event=] named {{inputreport}} at |device| using
+        {{HIDInputReportEvent}} with its {{HIDInputReportEvent/device}}
+        attribute initialized to |device|, its {{HIDInputReportEvent/reportId}}
+        attribute initialized to |reportId|, and its
+        {{HIDInputReportEvent/data}} attribute initialized to |data|.
+        </li>
+      </ol>
+      <section>
+        <h3>
+          <dfn>open()</dfn> method
+        </h3>
+        <p>
+          The {{HIDDevice/open()}} method steps are:
+        </p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          </li>
+          <li>If [=this=].{{HIDDevice/[[state]]}} is not `"closed"`, then
+          [=reject=] |promise| with a "{{InvalidStateError}}" {{DOMException}}
+          and return |promise|.
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[state]]}} to `"opening"`.
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[pendingOpenPromise]]}} to |promise|.
+          </li>
+          <li>Perform the following steps [=in parallel=]:
+            <ol>
+              <li>If [=this=].{{HIDDevice/[[pendingOpenPromise]]}} is not
+              |promise|, abort these steps.
+              </li>
+              <li>Invoke the operating system to open the HID device.
+              </li>
+              <li>If this fails for any reason, [=queue a global task=] on the
+              [=relevant global object=] of [=this=] using the [=HID device
+              task source=] to [=reject=] |promise| with a "{{NetworkError}}"
+              {{DOMException}} and abort these steps.
+              </li>
+              <li>Set [=this=].{{HIDDevice/[[state]]}} to `"opened"`.
+              </li>
+              <li>[=Queue a global task=] on the [=relevant global object=] of
+              [=this=] using the [=HID device task source=] to [=resolve=]
+              |promise| with `undefined`.
+              </li>
+            </ol>
+          </li>
+          <li>Return |promise|.
+          </li>
+        </ol>
+      </section>
+      <section>
+        <h3>
+          <dfn>close()</dfn> method
+        </h3>
+        <p>
+          The {{HIDDevice/close()}} method steps are:
+        </p>
+        <ol>
+          <li>Let |promise:Promise| be [=a new promise=].
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[state]]}} to `"closing"`.
+          </li>
+          <li>Let |pendingOpenPromise| be
+          [=this=].{{HIDDevice/[[pendingOpenPromise]]}}.
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[pendingOpenPromise]]}} to `null`.
+          </li>
+          <li>Let |pendingSendReportPromises| be
+          [=this=].{{HIDDevice/[[pendingSendReportPromises]]}}.
+          </li>
+          <li>Set [=this=].{{HIDDevice/[[pendingSendReportPromises]]}} to an
```

Comment:
I don't think this is necessary in specification text because rejecting a Promise queues a microtask.